WalletWasabi.Gui/Tabs/LegalDocumentsViewModel.cs

```diff
@@ -1,15 +1,58 @@
 using System;
 using System.IO;
+using System.Reactive;
+using System.Reactive.Linq;
 using Avalonia;
+using Avalonia.Logging;
 using Avalonia.Platform;
+using AvalonStudio.Extensibility;
+using AvalonStudio.Shell;
+using ReactiveUI;
+using Splat;
+using WalletWasabi.Gui.Helpers;
+using WalletWasabi.Gui.Models.TextResourcing;
 using WalletWasabi.Gui.ViewModels;
+using WalletWasabi.Legal;
 
 namespace WalletWasabi.Gui.Tabs
 {
 	public class LegalDocumentsViewModel : TextResourceViewModelBase
 	{
-		public LegalDocumentsViewModel() : base("Legal Documents", new Uri("avares://WalletWasabi.Gui/Assets/LegalDocuments.txt"))
+		public ReactiveCommand<Unit, Unit> AgreeClicked { get; set; }
+		public LegalDocuments LegalDoc { get; }
+
+		public bool IsAgreed { get; set; }
+
+		public LegalDocumentsViewModel(string content = null, LegalDocuments legalDoc = null) : base("Legal Documents", new TextResource() { FilePath = legalDoc?.FilePath, Content = content })
+		{
+			LegalDoc = legalDoc;
+			IsAgreed = content is null; // If content wasn't provided, then the filepath must had been provided. If the file exists, then it's agreed.
+
+			AgreeClicked = ReactiveCommand.CreateFromTask(async () =>
+			{
+				IsAgreed = true;
+				await LegalDoc.ToFileAsync(TextResource.Content);
+				Locator.Current.GetService<Global>().LegalDocuments = LegalDoc;
+				OnClose();
+			});
+
+			AgreeClicked
+				.ThrownExceptions
+				.ObserveOn(RxApp.TaskpoolScheduler)
+				.Subscribe(ex =>
+				{
+					NotificationHelpers.Error(ex.ToUserFriendlyString());
+					Logging.Logger.LogError(ex);
```

Comment:
Log the error first then display the UI notification. https://github.com/zkSNACKs/WalletWasabi/pull/2997#issue-365999512
Also add `using WalletWasabi.Logging;` to remove `Logging.`

Reply:
Cannot
![image](https://user-images.githubusercontent.com/9156103/73138953-c118cf00-403e-11ea-826d-8bdb124a60b9.png)